interop-testing/src/main/java/io/grpc/testing/integration/XdsTestServer.java

```diff
@@ -300,8 +302,44 @@ public void sendHeaders(Metadata responseHeaders) {
       };
       ServerCall.Listener<ReqT> noopListener = new ServerCall.Listener<ReqT>() {};
 
-      // sleep if instructed by rpc-behavior
-      for (String callBehavior : callBehaviors) {
+      int attemptNum = 0;
+      String attemptNumHeader = requestHeaders.get(ATTEMPT_NUM);
+      if (attemptNumHeader != null) {
+        try {
+          attemptNum = Integer.valueOf(attemptNumHeader);
+        } catch (NumberFormatException e) {
+          newCall.close(
+              Status.INVALID_ARGUMENT.withDescription(
+                  String.format(
+                      "Invalid format for grpc-previous-rpc-attempts header (%s)",
+                      attemptNumHeader)),
+              new Metadata());
+          return noopListener;
+        }
+      }
+
+      for (String callBehaviorEntry : callBehaviors) {
+        String callBehavior;
+        List<String> splitHeader = HEADER_HOSTNAME_SPLITTER.split(callBehaviorEntry);
+        if (splitHeader.size() > 1) {
+          String hostnameSpecifier = splitHeader[0];
+          if (hostnameSpecifier.startsWith(CALL_BEHAVIOR_HOSTNAME)) {
```

Comment:
I think this check really needs to be first, before the `splitHeader.size() > 1`. Just because it has a space somewhere doesn't mean it is related to hostname. And if it starts with hostname but doesn't have a space, we should fail.

Reply:
Done.